doc/nrf/templates/mesh_model_README.rst

```diff
@@ -0,0 +1,210 @@
+
+.. tip::
+   Describe the server model.
+   Start with a general purpose of the server model.
+   Provide basic information about the states, if the server model has any.
+
+   Additionally, list the model instances of the server model.
+   Say what they allow in relation to states and to each other, describe usage criteria and corner cases.
+
+Configuration*
+==============
+
+The following configuration parameters are associated with the XYZ server model:
+
+    * parameter 1
+    * parameter 2
+
+See :ref:`configure_application` for more information.
+
+.. tip::
+   List Kconfig options associated with the server model.
+   This section is optional, because not all models have Kconfig parameters that allow configuration.
+   Do not go into details here, but link to the parameter using the `:option:` reference.
+
+States
+======
+
+Generic Power On: `bt_mesh_template_location_state`
+    The Generic Power On state controls the default value of the disco light when the device powers up.
+    It can have the following values:
+
+        * `Value_1`: The disco light powers up and starts blinking.
+        * `Value_2`: The disco light powers up and starts with solid lighting.
+
+    Changes to the state should be exposed to the model through the XYZ callback.
+
+.. tip::
+   Describe each state using the definition list format (with header and indented definition).
+   Mention when it is used and why.
+   Provide information about variables, parameters, and values of the state.
+   If needed, describe the transition patterns and the callback used for exposing the changes to the model.
+
+Extended models
+===============
+
+.. tip::
+   Not all models have extended models.
+   If a model does not have any extended model, write *None* in this section.
+
+The XYZ Server extends the following models:
+
+    * Model 1
+    * Model 2
+
+The Generic Power On state is bound to the Generic OnOff state of the Model 1 through its power up behavior.
+No other state bindings are present, and the callbacks are forwarded to the application as they are.
+
+.. tip::
+   List the models that are extended by the model you are describing.
+   A model that extends another inherits the model's functionality and gives context to its states and messages.
+
+
+
+Persistent storage
+==================
+
+.. tip::
+   Not all models store information persistently.
+   If a model does not use persistent storage, write *None* in this section.
+
+The information about the Generic Power On state is stored persistently.
+
+.. tip::
+   Describe what information is stored persistently.
+   You can also specify where it is stored and what it is used for.
+
+
+API documentation
+=================
+
+| Header file: :file:`path/to/the/header/file`
+| Source file: :file:`path/to/the/source/file`
+
+.. tip::
+   Provide paths to the header and source files of the server model API.
+   Include also the doxygen group for the server model you are describing.
+
+
+----
```

Comment:
We're not using these lines anywhere else, and the pages are getting really long.
Should this be two (or more) separate subpages instead, one for each model, with the first section as parent?
Then you also don't need to duplicate the sections in the template for Server and Client, and we don't have the same headings on the same level twice.

Usually, the API documentation is the last thing on a page, and I'm not sure people will scroll down to look for more conceptual documentation.

Reply:
@trond-snekvik , what do you think about this proposal?
I was also considering using tabs, but you cannot use headings within tabs, so this won't work, unless you add tabs for Server and Client under each heading...

Reply:
Sounds like a good idea. Later models are going to get even more complex too, so we might as well do this split now